services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -134,16 +137,28 @@ protected void backchannelLogout(UserSessionModel userSession, String idToken) {
         UriBuilder logoutUri = UriBuilder.fromUri(getConfig().getLogoutUrl())
                 .queryParam("state", sessionId);
         logoutUri.queryParam("id_token_hint", idToken);
-        String url = logoutUri.build().toString();
-        try {
-            int status = SimpleHttp.doGet(url, session).asStatus();
-            boolean success = status >= 200 && status < 400;
-            if (!success) {
-                logger.warn("Failed backchannel broker logout to: " + url);
+
+        final String url = logoutUri.build().toString();
+        final HttpClient client = session.getProvider(HttpClientProvider.class).getHttpClient();
+        session.getTransactionManager().enlistAfterCompletion(new AbstractKeycloakTransaction() {
+            @Override
+            protected void commitImpl() {
+                try {
+                    int status = SimpleHttp.doGet(url, client).asStatus();
```

Comment:
@ahus1 Yes, I've tried using the original `SimpleHttp.doGet(url, session)` variant, which is implemented as

```
    public static SimpleHttp doGet(String url, KeycloakSession session) {
        return doGet(url, session.getProvider(HttpClientProvider.class).getHttpClient());
    }
```
and it failed with an NPE - the `session.getProvider(HttpClientProvider.class)` call returned `null` when called by the tx that is now enlisted at the end.

Reply:
I had a look at the code (didn't run it though), and can't why this happens. 

I'd say it would be good to have a code comment here describing why you did it this way including the error message you saw. 

If you'd log a GitHub issue on how to reproduce it, I'd be happy to analyze it separately so this can be simplified / doesn't fall on our feet later as it looks something might have been cleaned up before the `enlistAfterCompletion()` methods. 

WDYT?

Reply:
I think this happens because the session in this point is a proxy that isn't available/valid anymore when the second tx runs. The transaction runs as part of the overall commit handled by the filter, and at that point resteasy has already processed the original request and I think the proxies are no longer valid.

Reply:
Thanks, this resolves it for me. Thanks!